services/clusters-keeper/src/simcore_service_clusters_keeper/core/settings.py

```diff
@@ -139,10 +132,26 @@ class PrimaryEC2InstancesSettings(BaseCustomSettings):
 
     @validator("PRIMARY_EC2_INSTANCES_ALLOWED_TYPES")
     @classmethod
-    def check_valid_intance_names(cls, value):
+    def check_valid_instance_names(
+        cls, value: dict[str, EC2InstanceBootSpecific]
+    ) -> dict[str, EC2InstanceBootSpecific]:
         # NOTE: needed because of a flaw in BaseCustomSettings
         # issubclass raises TypeError if used on Aliases
-        parse_obj_as(tuple[InstanceTypeType, ...], value)
+        if all(parse_obj_as(InstanceTypeType, key) for key in value):
```

Comment:
ok I think there is some confusion here.
1. First the flaw I talk about above concerns BaseCustomSettings. If I use a InstanceTypeType which is a TypeAlias in a Settings object this does not work well.
2. what you are commenting on was changed because the type went from a list[str] to dict[str, EC2InstanceSpecs] or similar. ```parse_obj_as``` works totally well.